modules/osdk-updating-v125-to-v128.adoc

```diff
@@ -0,0 +1,150 @@
+endif::[]
+ifeval::["{context}" == "osdk-helm-updating-projects"]
+:helm:
+:type: Helm
+endif::[]
+ifeval::["{context}" == "osdk-hybrid-helm-updating-projects"]
+:hybrid:
+:type: Hybrid Helm
+endif::[]
+ifeval::["{context}" == "osdk-java-updating-projects"]
+:java:
+:type: Java
+endif::[]
+
+:_content-type: PROCEDURE
+[id="osdk-upgrading-projects_{context}"]
+= Updating {type}-based Operator projects for Operator SDK {osdk_ver}
+
+The following procedure updates an existing {type}-based Operator project for compatibility with {osdk_ver}.
+
+ifdef::golang[]
+[NOTE]
+====
+The `go/v3` plugin is being deprecated by Kubebuilder. As a result, the Operator SDK will also be migrating to the `go/v4` plugin in a future release.
+====
+endif::[]
+
+.Prerequisites
+
+* Operator SDK {osdk_ver} installed
+* An Operator project created or maintained with Operator SDK {osdk_ver_n1}
+
+.Procedure
+
+ifdef::ansible[]
+. Update your Makefile's `run` target to the following:
++
+[source,make]
+----
+.PHONY: run
+ANSIBLE_ROLES_PATH?="$(shell pwd)/roles"
+run: ansible-operator ## Run against the configured Kubernetes cluster in ~/.kube/config
+    $(ANSIBLE_OPERATOR) run
+----
+
+. To upgrade the `kubernetes.core` collection to v2.4.0, replace the following in your project's `requirements.yaml` file:
++
+[source,yaml]
+----
+  - name: kubernetes.core
+    version: "2.3.1"
+----
++
+with:
++
+[source,yaml]
+----
+  - name: kubernetes.core
+    version: "2.4.0"
+----
+
+. You must upgrade your project to support Kubernetes 1.26:
+endif::[]
+ifndef::ansible[]
+* You must upgrade your project to support Kubernetes 1.26:
+endif::[]
+
+.. Modify your `go.mod` file to upgrade the dependencies:
++
+[source,go]
+----
+k8s.io/api v0.26.2
+k8s.io/apiextensions-apiserver v0.26.2
+k8s.io/apimachinery v0.26.2
+k8s.io/cli-runtime v0.26.2
+k8s.io/client-go v0.26.2
+k8s.io/kubectl v0.26.2
+sigs.k8s.io/controller-runtime v0.14.5
+sigs.k8s.io/controller-tools v0.11.3
+sigs.k8s.io/kubebuilder/v3 v3.9.1
+----
+
+.. Download the latest dependencies by running the following command:
++
+[source,terminal]
+----
+$ go mod tidy
+----
+
+.. With the update to Kubebuilder 3.9.1 in Operator SDK {osdk_ver}, `go/v3` projects are now scaffolded with `kube-rbac-proxy` v0.13.1. Upgrade the version of `kube-rbac-proxy` in the following files in your project:
++
+--
+* `config/default/manager_auth_proxy_patch.yaml`
+* `bundle/manifests/memcached-operator.clusterserviceversion.yaml`
+--
++
+[source,diff]
+----
+    - gcr.io/kubebuilder/kube-rbac-proxy:v0.13.0
+    + gcr.io/kubebuilder/kube-rbac-proxy:v0.13.1
+----
+
+ifdef::golang[]
+.. Modify your Makefile with the following changes:
+
+... Change the `ENVTEST_K8S_VERSION` field from `1.25` to `1.26`.
+... Change the `build` target from `generate fmt vet` to `manifests generate fmt vet`:
++
+[source,diff]
+----
+    - build: generate fmt vet ## Build manager binary.
+    + build: manifests generate fmt vet ## Build manager binary.
+----
+endif::[]
```

Comment:
Why the ansible & helm & hybrid & Java based update doc is the same with go type operator? They are different.

Reply:
Adjusted the `golang` conditionals.